pivoting/Pivoting_Study.md

```diff
@@ -1,30 +1,110 @@
 # Pivoting Study
-## Steps
-* Fetch:
-    * Nordix/metal3-dev-env:test/pivoting-kashif branch
-    * Nordix/cluster-api-provider-baremetal:test/pivoting-kashif
-    * Nordix/cluster-api-provider-baremetal:test/pivoting-kashif
-** Notes: These branches might not be rebased on current upstream branch. This is my working branch.
+Pivoting is the process of moving controllers and objects from one k8s cluster to another. The purpose of this document is to describe the process, the challenges encountered and future works.
 
-* Make sure you have the ```provider-components.yaml``` in ```cluster-api-provider-baremetal\examples\_out``` directory. ```make deploy ``` would create this. 
+The following controllers and objects are relevant in the in the context of baremetal based kubernetes deployment.
 
-* Make a copy of it and create ```provider-components-target.yaml```. Assuming that, you are creating the target clusters in ```ubuntu```, change the ```eth2``` to ```enp1s0``` in ```config-map``` section of ```provider-components-target.yaml```.
+**Controllers:** CAPI, CAPBK, CAPBM and BMO
+**Main Objects:** Cluster, Machine, MachineSet, MachineDeployment, 
+BareMetalCluster, BareMetalMachine and BareMetalHost
+**Related Objects:** Cluster Secrets, BareMetalHost Secrets, Configmaps
 
-* Do the following steps:
+ The following table lists the controllers with their versions used in this process.
+
+| Repo        | Version           | Branch  |
+| ------------- |:-------------:| -----:|
+| kubernetes      | 1.17.0 | master |
+| CAPI      | v1alpha2      |   release-0.2 |
+| CAPBM |    v1alpha2      |   master |
+| BMO |    v1alpha1      |   master |
+
+The clusterctl that we have developed specifically for pivoting BareMetal based deployments is based on CAPI-clusterctl release-0.2
+
+### Pivoting process
+
+- The source of truth as of ```release-0.2``` is the provider components yaml file. 
+- ```clusterctl``` binary takes the source and target cluster kubeconfig and the provider components file as arguements. 
```

Comment:
binary takes the source and target clusters' kubeconfig and the provider components file as arguments.

Reply:
Resolved